libraries/AP_HAL_SITL/SITL_State.cpp

```diff
@@ -75,13 +75,13 @@ void SITL_State::_sitl_setup(const char *home_str)
 #endif
     fprintf(stdout, "Starting SITL input\n");
 
+    _sitl = SITL::SITL::instance();
     // find the barometer object if it exists
-    _sitl = (SITL::SITL *)AP_Param::find_object("SIM_");
-    _barometer = (AP_Baro *)AP_Param::find_object("GND_");
-    _ins = (AP_InertialSensor *)AP_Param::find_object("INS_");
-    _compass = (Compass *)AP_Param::find_object("COMPASS_");
+    _barometer = AP_Baro::instance();
+    _ins = AP_InertialSensor::get_instance();
```

Comment:
AP_InertialSensor was already a singleton hence the different function naming. I wasn't sure whether I should update it accordingly to match the rest. Let me know and I'll quickly update it.